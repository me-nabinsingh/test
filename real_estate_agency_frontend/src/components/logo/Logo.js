import React from 'react';
import { Link } from "react-router-dom";
import logo from '../../assets/images/logo.svg';
import PropTypes from 'prop-types';


const Logo = ({width, height}) => {
    return ( 
        <Link className="py-2" to="/">
            <img  alt="" src={logo} width={width} height={height} />
        </Link>
     );
}

Logo.defaultProps = {
    width: 50,
    height: 50
}

Logo.propTypes = {
    width: PropTypes.number,
    height: PropTypes.number
}
 
export default Logo;