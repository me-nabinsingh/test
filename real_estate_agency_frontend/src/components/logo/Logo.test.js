import React from 'react';
import Logo from './Logo';
import {Router } from "react-router-dom";
import renderer from 'react-test-renderer';
import history from "../../util/history";


const width = 30, height =  30;

describe('<Logo />', () => {
    it("doesn't break if width and hight are not passed", () => {
        const component = renderer.create(
            <Router history={history}>
                <Logo />
            </Router>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })


    it('renders without crashing', () => {
        
        const component = renderer.create(
            <Router history={history}>
                <Logo width={width} height={height}/>
            </Router>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});