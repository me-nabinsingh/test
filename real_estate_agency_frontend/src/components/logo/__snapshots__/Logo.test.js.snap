// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> doesn't break if width and hight are not passed 1`] = `
<a
  className="py-2"
  href="/"
  onClick={[Function]}
>
  <img
    alt=""
    height={50}
    src="logo.svg"
    width={50}
  />
</a>
`;

exports[`<Logo /> renders without crashing 1`] = `
<a
  className="py-2"
  href="/"
  onClick={[Function]}
>
  <img
    alt=""
    height={30}
    src="logo.svg"
    width={30}
  />
</a>
`;
